### PR TITLE
Complete integration of Rust BlockValidator

### PR DESCRIPTION
**Pull request by boydjohnson**
~~This PR is built on #1828 .~~

### CODE REVIEW

File: validator/src/journal/block_manager.rs [hunk 1]
```diff
@@ -430,6 +430,14 @@ impl BlockManager {
         Box::new(GetBlockIterator::new(Arc::clone(&self.state), block_ids))
     }
 
+    pub fn get_block_from_blockstore(
```
Comment (peterschwarz) [inline validator/src/journal/block_manager.rs]: block seems redundent; `get_from_blockstore` would be fine

File: validator/src/journal/block_validator.rs [hunk 2]
```diff
@@ -217,10 +220,14 @@ where
         transaction_index: T,
         view_factory: StateViewFactory,
     ) -> Self {
-        let (tx, rcv) = channel::<(Block, Sender<BlockValidationResult>)>();
+        let (tx1, rcv1) = channel::<(Block, Sender<BlockValidationResult>)>();
+        let (tx2, rcv2) = channel::<(Block, Sender<BlockValidationResult>)>();
         BlockValidator {
-            tx,
-            rx: Some(rcv),
+            tx1,
```
Comment (peterschwarz) [inline validator/src/journal/block_validator.rs]: Why not just a vec of tuples?

Comment (boydjohnson) [reply validator/src/journal/block_validator.rs]: Makes sense. Then there can be a configurable number of background threads.

File: validator/src/journal/block_manager.rs (hunk 1 above)
Comment (boydjohnson) [reply validator/src/journal/block_manager.rs]: Sounds good.

**Pull request merged**